src/expandoracommon/parseevent.h

```diff
@@ -49,12 +50,16 @@ class ParseEvent : public ListCycler<Property *, std::deque<Property *> >
   const std::deque<QVariant> & getOptional() const {return optional;}
   uint getMoveType() const {return moveType;}
   uint getNumSkipped() const {return numSkipped;}
-  
+
+  void setParsed();
+  bool waitForParsed(int timeout=-1);
```

Comment:
Wouldn't it be safer to set this to some more reasonable value? What if the server freezes? Won't this lock up the entire thread?